TestPlaybooks/playbook-TestExtractHTMLTables.yml

```diff
@@ -0,0 +1,445 @@
+    nexttasks:
+      '#none#':
+      - "2"
+    scriptarguments:
+      all:
+        simple: "yes"
+      key: {}
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 245,
+          "y": 195
+        }
+      }
+  "2":
+    id: "2"
+    taskid: 6cc1640f-24df-4b6a-8cf2-5d2daaac5c57
+    type: regular
+    task:
+      id: 6cc1640f-24df-4b6a-8cf2-5d2daaac5c57
+      version: -1
+      name: ExtractHTMLTables with single one column table with header
+      description: |-
+        Find tables inside HTML and extract the contents into objects using the following logic:
+        - If table has a single column, just create an array of strings from the values
+        - If table has 2 columns and has no header row, treat the first column as key and second as value and create a table of key/value
+        - If table has a header row, create a table of objects where attribute names are the headers
+        - If table does not have a header row, create table of objects where attribute names are cell1, cell2, cell3...
+      scriptName: ExtractHTMLTables
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "3"
+    scriptarguments:
+      html:
+        simple: <html><body><table><tr><th>kuku</th></tr><tr><td>a</td></tr><tr><td>1</td></tr></table></body></html>
+      indexes: {}
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 245,
+          "y": 370
+        }
+      }
+  "3":
+    id: "3"
+    taskid: 9f902e22-b34e-4559-8238-8752c1afda3e
+    type: condition
+    task:
+      id: 9f902e22-b34e-4559-8238-8752c1afda3e
+      version: -1
+      name: Both values exist in array
+      description: Check if a given value exists in the context. Will return 'no'
+        for empty empty arrays. To be used mostly with DQ and selectors.
+      scriptName: Exists
+      type: condition
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#default#':
+      - "5"
+      "yes":
+      - "6"
+    scriptarguments:
+      value:
+        simple: '${HTMLTables=>val[0] == ''a'' && val[1] == ''1'' ? ''x'' : ''''}'
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 245,
+          "y": 545
+        }
+      }
+  "4":
+    id: "4"
+    taskid: 4861114d-746b-4d17-8a2f-3a730474a3eb
+    type: regular
+    task:
+      id: 4861114d-746b-4d17-8a2f-3a730474a3eb
+      version: -1
+      name: Close
+      scriptName: CloseInvestigation
+      type: regular
+      iscommand: false
+      brand: ""
+    scriptarguments:
+      notes:
+        simple: All good
+      reason: {}
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 592.5,
+          "y": 2295
+        }
+      }
+  "5":
+    id: "5"
+    taskid: 50da60bb-61ac-4ad6-8f89-8f43eb32cb9f
+    type: regular
+    task:
+      id: 50da60bb-61ac-4ad6-8f89-8f43eb32cb9f
+      version: -1
+      name: Bad
```

Comment:
This Bad task is regular manual task, so if the playbook will reach here, nothing will happen. Playbook should be in error state in order to fail.
So you can add automation script like Equals 0=1 (always false)

Reply:
Will do - did not remember if we actually need to fail